### PR TITLE
Add one documentation about transaction

### PR DESCRIPTION
**Pull request by Pana**

### CODE REVIEW

File: docs/sending-tx/en/sending-tx-ultimate-guide.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+2. `'Invalid parameters: tx', 'data': '"Transaction 0x2004b0aea956e8cfad601cd6daad5630c1a95624bad446d1966895973325136c is discarded due to a too stale nonce` This indicates that the same nonce used has been used in history and cannot be used again.
+3. `Sending transactions to invalid address. The first four bits must be 0x0 (built-in/reserved), 0x1 (user-account), or 0x8 (contract).` This indicates that the receiving address of the transaction is wrong.
+4. `Transaction {:?} is discarded due to in too distant future` This indicates that the transaction is using a nonce that is too large for the account's current nonce
+5. `tx already exist` This indicates that the transaction already exists.
+
+Sometimes, the `data` of Response will also contain some error information. （This field is a hex-encoded string. The parsing method is: hex -> buffer -> UTF8 string）
+
+### Stucked in the transaction pool
```
Comment (Thegaram) [inline docs/sending-tx/en/sending-tx-ultimate-guide.md:180]: > Stucked in the transaction pool

*"Stuck in the transaction pool"*

File: docs/sending-tx/en/sending-tx-ultimate-guide.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+The transaction is sent successfully but not be packaged. There are two possible reasons:
+
+1. The skipped nonce is used, and the transaction of the previous nonce has not been generated or executed.
+2. The account sending this transaction does not have enough balance to pay the transaction.
+
+### Executed failed
```
Comment (Thegaram) [inline docs/sending-tx/en/sending-tx-ultimate-guide.md:186]: > Executed failed

*"Execution failed"*

File: docs/sending-tx/en/sending-tx-ultimate-guide.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+Execution failure is usually due to an error occurred during the execution process of the contract, which then caused the transaction failure. Such errors are mostly caused by contract execution failures or errors returned when estimating gas cost through the estimate interface.
```
Comment (Thegaram) [inline docs/sending-tx/en/sending-tx-ultimate-guide.md:187]: > error occurred

*"error that occurred"*

File: docs/sending-tx/en/sending-tx-ultimate-guide.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+You can check the specific reason for the transaction failure in the `txExecErrorMsg` under receipt:
+
+1. `VmError(OutOfGas)` The transaction specified gas fee is not enough
+2. `VmError(ExceedStorageLimit)` The transaction specified upper-limit storage is not enough
+3. `NotEnoughCash` Insufficient user balance
+4. `Vm reverted, Reason provided by the contract: xxxx` The contract execution failed with details provided. 
+5. `VmError(BadInstruction` Contract deployment failed
```
Comment (Thegaram) [inline docs/sending-tx/en/sending-tx-ultimate-guide.md:194]: > `VmError(BadInstruction`

`VmError(BadInstruction)`

File: docs/sending-tx/en/sending-tx-ultimate-guide.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+6. `Vm reverted, ` The contract execution failed with no details provided.
```
Comment (Thegaram) [inline docs/sending-tx/en/sending-tx-ultimate-guide.md:195]: > `Vm reverted, `

`Vm reverted`

File: docs/sending-tx/en/sending-tx-ultimate-guide.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+
+## Difference between Conflux and Ethereum
```
Comment (Thegaram) [inline docs/sending-tx/en/sending-tx-ultimate-guide.md:198]: > Difference

*"Differences"*

File: docs/sending-tx/en/sending-tx-ultimate-guide.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+Compared to Ethereum, transactions through Conflux have three more fields: `chainId`, `storageLimit`, and `epochNumber`. The rlp encoding assembly method of the transaction is different. In addition, the v value signed by ecdsaSign will not be specifically modified in Conflux; while In Ethereum, there will be some special treatments to the v value.
```
Comment (Thegaram) [inline docs/sending-tx/en/sending-tx-ultimate-guide.md:199]: Maybe add:

*"Finally, while Ethereum nodes directly reject transactions if the sender has insufficient balance, Conflux nodes accept such transactions but only pack them info blocks once the user's balance can cover it."*

**Comment by Pana** [comment]
Add docs: cfx role

File: docs/introduction/en/basics.md [hunk 8]
```diff
@@ -18,6 +18,21 @@ To incentivize the maintenance of the Conflux network and charge users for consu
 | 10^12         | uCFX          |
 | 10^18         | Conflux(CFX)  |
 
+As a native token of the Conflux network, CFX plays a very important role in the stability of the system.
+
+First of all, CFX is used to pay transaction fees and to store collateral. Transactions are packaged and executed by miners. In order to incentivize miners to keep participating in the bookkeeping and provide sufficient security for the network, each transaction is charged a (gas) fee calculated as gasPrice * max(gasLimit * 3/4, gasUsed), which is paid by CFX. In addition, if the execution of a transaction takes up new storage space, a portion of the CFX is pledged for the use of the storage space (the pledged CFX can be released under certain condition).
+
+Conflux introduces a [staking mechanism (Protocol Specification chapter 8.3)](https://confluxnetwork.org/files/Conflux_Protocol_Specification_20201020.pdf). This mechanism not only provides a better way to pay for storage usage (as opposed to "pay once, occupy forever") but also serves to construct the voting rights of Conflux decentralized governance. The annualized income of collateralized CFX is around 4%.
```
Comment (Thegaram) [inline docs/introduction/en/basics.md:25]: This part is confusing, it mixes staking through internal contract and storage collateral together.

**Review by Thegaram — APPROVED** [review]

**Pull request merged**